elmslie-store-persisting/src/main/java/vivid/money/elmslie/storepersisting/RetainedStoreHolder.kt

```diff
@@ -0,0 +1,59 @@
+package vivid.money.elmslie.storepersisting
+
+import android.os.Bundle
+import androidx.lifecycle.*
+import androidx.savedstate.SavedStateRegistryOwner
+
+fun <R : ClearableStoreHolder<*, *, *>> createRetainedStoreHolderLazy(
```

Comment:
I'd suggest to annotate it with `@MainThread` since we're forcing no thread safety here

Reply:
Annotated